src/Tempest/Http/src/GenericRouter.php

```diff
@@ -45,7 +43,7 @@ public function dispatch(Request|PsrRequest $request): Response
             $request = map($request)->with(RequestToPsrRequestMapper::class);
         }
 
-        $matchedRoute = $this->matchRoute($request);
+        $matchedRoute = $this->routeMatcher->match($request);
```

Comment:
I don't have a strong opinion on it, but maybe we could move the registration of `MatchedRoute` in the container to `RouteMatcher` as well? Currently, the router still does it on line 52.

Reply:
Good point.